File: lib/qt4/TypeListComboBox.cpp
/*
 * Copyright (c) 2013-2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <string>
#include <vector>

#include <pkmn/lists.hpp>
#include <pkmn/qt4/TypeListComboBox.hpp>

namespace pkmn
{
    namespace qt4
    {
        TypeListComboBox::TypeListComboBox(const uint16_t gen, QWidget* parent):
            QComboBox(parent)
        {
            std::vector<pkstring> types_vec;
            get_type_list(types_vec, gen);

            for(int i = 0; i < types_vec.size(); i++)
                addItem(QString::fromUtf16(types_vec[i]), QVariant(i));
        }
    } /* namespace qt4 */
} /* namespace pkmn */

File: lib/swig/java/ComboBoxes/TypeListComboBox.java
/*
 * Copyright (c) 2013-2015 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.*;

/**
 * A ComboBox that lists all types present in the given generation.
 *
 * @param gen generation whose types to return
 */
public class TypeListComboBox extends JComboBox<String>
{
    public TypeListComboBox(int gen)
    {
        super();

        StringVector typeVec = new StringVector();
        Lists.getTypeList(typeVec, gen);
        for(int i = 0; i < typeVec.size(); i++)
        {
            addItem(typeVec.get(i));
        }

        setEditable(false);
    }
}
